Orange/widgets/data/owdatasets.py

```diff
@@ -141,6 +141,16 @@ def __init__(self):
         self.infolabel = QLabel(text="Initializing...\n\n")
         box.layout().addWidget(self.infolabel)
 
+        gui.widgetLabel(self.mainArea, "Filter")
+        self.filterLineEdit = QLineEdit(
+            textChanged=self.filter
+        )
+        self.completer = TokenListCompleter(
+            self, caseSensitivity=Qt.CaseInsensitive
+        )
+        self.filterLineEdit.setCompleter(self.completer)
```

Comment:
Oops, you are right. I started moving functionality from GEO Datasets, but then changed to a simpler version. No caching from `MySortFilterProxyModel` and no completion. It actually seemed good enough for now. I haven't tested when caching would become important, but there are no performance problems for now (and probably won't be in the near future for a manually curated list of data sets). We might want to consider enabling/finishing completion (since it is already done in bioinf.), however, we can also go with a bare-bones approach (and remove moving that class for now) => less code, easier to maintain?
@ales-erjavec are the two related? Does adding a completer require the use of caching (probably should not be the case) or was there another reason for that in GEO (more data sets, automatically synced with a big external repository)?

Reply:
> Are the two related? Does adding a completer require the use of caching.

No.